aws-s3/scripts-ccao-data-raw-us-east-1/spatial-corner_lot.R

```diff
@@ -0,0 +1,360 @@
+
+  cross_corner_number <- nrow(cross_corner)
+
+  aspect_ratio <- max(cross$aspect_ratio, na.rm = TRUE)
+
+  # Check for an edge case where two corner segments are actually on opposing
+  # sides of the parcel
+  back_front_indicator <- ifelse(
+    touching_street_number == 2 &
+      (!is_empty(touching_unit_street[[1]]) & !is_empty(touching_unit_street[[3]])) |
+      (touching_street_number == 2 & !is_empty(touching_unit_street[[2]]) & !is_empty(touching_unit_street[[4]])),
+    TRUE,
+    FALSE
+  )
+
+  back_front_indicator <- as.numeric(back_front_indicator)
+
+  # Check for an edge case where the same road has two representations in the
+  # network, and so gets erroneously counted as two roads
+  double_road_indicator <- ifelse(
+    touching_street_number > 1 &
+      (is_empty(touching_unit_street[[1]]) & is_empty(touching_unit_street[[2]]) & is_empty(touching_unit_street[[3]])) |
+      (is_empty(touching_unit_street[[2]]) & is_empty(touching_unit_street[[3]]) & is_empty(touching_unit_street[[4]])) |
+      (is_empty(touching_unit_street[[1]]) & is_empty(touching_unit_street[[3]]) & is_empty(touching_unit_street[[4]])) |
+      (is_empty(touching_unit_street[[1]]) & is_empty(touching_unit_street[[2]]) & is_empty(touching_unit_street[[4]])),
+      TRUE,
+    FALSE
+  )
+
+  double_road_indicator <- as.numeric(double_road_indicator)
+
+  # Step 6: Check if the parcel meets all the necessary conditions to be counted
+  # as a corner
+  if (all(touching_street_number >= 2) & all(cross_corner_number >= 1) & all(aspect_ratio < 30) & back_front_indicator == 0 & double_road_indicator == 0){
+    return(TRUE)
+  } else {
+    return(FALSE)
+  }
+}
+
+# Parcel data needs to be chunked in order to reduce memory use. Set the
+# size of the chunks based on the capacity of the compute environment
+chunk_size <- 5000
+
+# Calculate the total number of chunks
+num_chunks <- ceiling(nrow(parcels) / chunk_size)
+
+# Create a list to store the results
+results_list <- vector("list", length = num_chunks)
+
+# Iterate through chunks of parcels and apply the corner indicator function
+results_list <- map(1:num_chunks, function(chunk_number) {
+  start_index <- (chunk_number - 1) * chunk_size + 1
+  end_index <- min(chunk_number * chunk_size, nrow(parcels))
+
+  parcel_slice <- parcels %>%
+    slice(start_index:end_index) %>%
+    st_transform(3435)
+
+  result <- numeric(nrow(parcel_slice))
+
+  for (x in 1:nrow(parcel_slice)) {
+    result[x] <- parcel_is_corner(x, parcel_slice, parcels_buffered, network_trans)
+    cat("Chunk:", chunk_number, "/", num_chunks, "- Iteration:", x, "/", nrow(parcel_slice), "\n")
+  }
+
+  return(result)
+})
+
+# Combine the results from each chunk
+final_result <- unlist(results_list)
+
+final <- cbind(parcels, final_result)
+
+# Write the parcel results to disk
+file_name <- paste0(township, "_11_20.shp")
+
+directory <- file.path(
+  "aws-s3", "scripts-ccao-data-raw-us-east-1", "spatial-corner_lot-raw", township
+)
```

Comment:
This is the online line that has changed since https://github.com/ccao-data/data-architecture/pull/260:

```diff
-directory <- file.path("Base Data", township)
+directory <- file.path(
+  "aws-s3", "scripts-ccao-data-raw-us-east-1", "spatial-corner_lot-raw", township
+)
```

This new directory path should keep the output a little bit more organized, since it will no longer show up in the root of the `data-architecture` repo.

Reply:
And, should I move existing files into this location?